Gnome 3.34

hedning opened the pull request:
<!-- Nixpkgs has a lot of new incoming Pull Requests, but not enough people to review this constant stream. Even if you aren't a committer, we would appreciate reviews of other PRs, especially simple ones like package updates. Just testing the relevant package/service and leaving a comment saying what you tested, how you tested it and whether it worked would be great. List of open PRs: <https://github.com/NixOS/nixpkgs/pulls>, for more about reviewing contributions: <https://hydra.nixos.org/job/nixpkgs/trunk/manual/latest/download/1/nixpkgs/manual.html#sec-reviewing-contributions>. Reviewing isn't mandatory, but it would help out a lot and reduce the average time-to-merge for all of us. Thanks a lot if you do! -->

[Jobset Evals](https://hydra.nixos.org/jobset/nixpkgs/gnome#tabs-evaluations)

**as of 2019-09-20**
https://hydra.nixos.org/eval/1543560?compare=1542887

###### Blocking issues
- [x] update dash to dock (no release yet...)
- [x] `totem` and `gnome-music` crashes on startup with `grilo-plugins-0.3.10`
https://gitlab.gnome.org/GNOME/tracker-miners/issues/81
- [x] ~wingpanel-indicator-datetime isn't ported to libecal-2.0~ workaround https://github.com/NixOS/nixpkgs/pull/68608/commits/9b061db3bf9a4002cddffee0c942c56f2f23cb0f
https://github.com/elementary/wingpanel-indicator-datetime/issues/122
  Without this Pantheon test will fail and block the channel.

###### TODO

- [x] webkitgtk now require gcc >= 7, which means we now hit this: https://github.com/NixOS/nixpkgs/pull/47949#discussion_r223853101 ([build log](https://gist.github.com/hedning/e3611ffbd9bf4a928c5c6d52afde9eca))
- [x] systemd session: just needed to manually specify that gnome-session wants gsd services (as `systemd.packages` don't pick .wants directories for some reason) 
- [x] GDM is a bit buggy, in particular we hit this when starting on vt7: https://gitlab.gnome.org/GNOME/gdm/issues/515 (when starting on vt1 it seems like gdm gets killed quickly and I need to go to another vt and back to vt1 for it to be spawned again). Fixed by making sure gdm starts after tty1 (having gdm start on tty7 messes tty switching for some reason though). ~Edit: breaks [nixos-rebuild test/switch](https://github.com/NixOS/nixpkgs/issues/21439)~ 
  ~Pretty sure that starting gdm on tty1 isn't what's causing nixos-rebuild switch to restart gdm (see eg. https://github.com/NixOS/nixpkgs/issues/44344)~
- [x] make it possible to run mutter using realtime scheduling: https://github.com/NixOS/nixpkgs/issues/68318#issuecomment-530878341
- [x] enable new gnome-shell `extension_tool`: https://gitlab.gnome.org/GNOME/gnome-shell/blob/master/meson.build#L135
- [x] enable mutter's new profiler: https://gitlab.gnome.org/GNOME/mutter/blob/master/meson.build#L273
- [x] seahorse fails to build with new libsecret: https://gitlab.gnome.org/GNOME/seahorse/issues/242
- [x] elementary-calendar isn't ported completely to libecal-2.0 https://github.com/elementary/calendar/pull/431 https://github.com/elementary/calendar/issues/404
- [x] build webkitgtk with clang
- [x] gnome-initial-setup shows up in gdm (note in lightdm it doesn't happen)
- [ ] disable gnome-initial-setup in tests or adapt to include it
- [x] update extensions
- [ ] fix libpeas .pc upstream https://gitlab.gnome.org/GNOME/libpeas/merge_requests/16

###### Nice to have

- [ ] enable mutter's installed tests
- [ ] seahorse should use meson's gpgme-config detection https://github.com/mesonbuild/meson/pull/5213
- [x] correct upstream atk .pc file https://gitlab.gnome.org/GNOME/atk/merge_requests/30
- [x] gnome-session systemd always has debug output https://gitlab.gnome.org/GNOME/gnome-session/merge_requests/23

###### Things done

I've gotten my own system up and running without any major issues :)

closes https://github.com/NixOS/nixpkgs/issues/68318

worldofpeace committed:
pantheon.elementary-session-settings: add default settings to config dirs

Also don't use placeholder in the executable.

jtojnar committed:
gnome-firmware-updater: init at 3.34.0

Co-authored-by: worldofpeace <worldofpeace@protonmail.ch>

hedning committed:
glib: 2.60.7 -> 2.62.0

Co-authored-by: Jan Tojnar <jtojnar@gmail.com>

hedning committed:
at-spi2-core: 2.32.1 → 2.34.0

hedning committed:
at-spi2-atk: 2.32.0 → 2.34.0

hedning committed:
evolution-data-server: 3.32.4 → 3.34.0

hedning committed:
folks: 0.12.1 → 0.13.1

hedning committed:
gcr: 3.28.1 → 3.33.4

hedning committed:
gnome3.gnome-online-accounts: 3.32.0 → 3.34.0

hedning committed:
gnome-photos: 3.32.1 → 3.34.0

hedning committed:
gnome-builder: 3.32.4 → 3.34.0

* add pcre2
* don't override stdenv

Co-authored-by: worldofpeace <worldofpeace@protonmail.ch>

hedning committed:
gnome-usage: 3.32.0 → 3.33.2

hedning committed:
gvfs: 1.40.2 → 1.42.0

Non-gnome build now require gnome-desktop-schemas

hedning committed:
gnome3.accerciser: 3.32.3 → 3.34.0

hedning committed:
gnome3.cheese: 3.32.1 → 3.34.0

hedning committed:
gnome3.dconf-editor: 3.32.0 → 3.34.1

hedning committed:
gnome3.eog: 3.32.2 → 3.34.0

hedning committed:
gnome3.four-in-a-row: 3.32.0 → 3.34.0

hedning committed:
gnome3.gdm: 3.32.0 → 3.34.0

- disable custom initial vt as it's buggy

hedning committed:
gnome3.geary: 3.32.2 -> 3.34.0

hedning committed:
gnome3.evolution: 3.32.4 → 3.34.0

hedning committed:
gnome3.gedit: 3.32.2 → 3.34.0

hedning committed:
gnome3.gjs: 1.56.2 → 1.58.0

- No longer links to gkt3

hedning committed:
gnome3.gnome-bluetooth: 3.32.1 → 3.34.0

hedning committed:
gnome3.gnome-calculator: 3.32.2 → 3.34.0

hedning committed:
gnome3.gnome-books: 3.32.0 → 3.34.0

hedning committed:
gnome3.gnome-calendar: 3.32.2 → 3.34.0

hedning committed:
gnome3.gnome-clocks: 3.32.0 → 3.34.0

hedning committed:
gnome3.gnome-chess: 3.32.0 → 3.34.0

hedning committed:
gnome3.gnome-contacts: 3.32.1 → 3.34

hedning committed:
gnome3.gnome-control-center: 3.32.2 → 3.34.0.1

hedning committed:
gnome3.gnome-desktop: 3.32.2 → 3.34.0

hedning committed:
gnome3.gnome-flashback: 3.32.0 → 3.34.0

Now longer has a gnome-flashback-init component.

hedning committed:
gnome3.gnome-documents: 3.32.0 → 3.33.90

hedning committed:
gnome3.gnome-disk-utility: 3.32.1 → 3.34.0

hedning committed:
gnome3.gnome-font-viewer: 3.32.0 → 3.34.0

hedning committed:
gnome3.gnome-logs: 3.32.1 → 3.34.0

hedning committed:
gnome3.gnome-klotski: 3.32.0 → 3.34.0

hedning committed:
gnome3.gnome-mahjongg: 3.32.0 → 3.34.0

hedning committed:
gnome3.gnome-maps: 3.32.2.1 → 3.34.0

hedning committed:
gnome3.gnome-mines: 3.32.2 → 3.34.0

hedning committed:
gnome3.gnome-music: 3.32.2 → 3.34.0

hedning committed:
gnome3.gnome-nibbles: 3.32.0 → 3.34.0

hedning committed:
gnome3.gnome-robots: 3.32.0 → 3.34.0

hedning committed:
gnome3.gnome-getting-started-docs: 3.32.2 → 3.34.0

hedning committed:
gnome3.gnome-panel: 3.32.0 → 3.34.0

hedning committed:
gnome3.gnome-screenshot: 3.32.0 → 3.33.90

hedning committed:
gnome3.gnome-session: 3.32.0 → 3.34.0

* adds systemd sessions

hedning committed:
gnome3.gnome-settings-daemon: 3.32.1 → 3.34.0

- fixup build

hedning committed:
gnome3.gnome-shell-extensions: 3.32.1 → 3.34.0

hedning committed:
gnome3.gnome-sudoku: 3.32.0 → 3.34.0

hedning committed:
gnome3.gnome-tetravex: 3.32.0 → 3.34.0

hedning committed:
gnome3.gnome-terminal: 3.32.2 → 3.34.0

* add pcre2

Co-authored-by: worldofpeace <worldofpeace@protonmail.ch>

hedning committed:
gnome3.gnome-taquin: 3.32.0 → 3.34.0

hedning committed:
gnome3.gnome-tweaks: 3.32.0 → 3.34.0

hedning committed:
gnome3.gnome-user-share: 3.32.0.1 → 3.34.0

hedning committed:
gnome3.gnome-user-docs: 3.32.3 → 3.34.0

hedning committed:
gnome3.gnome-weather: 3.32.2 → 3.34.0

hedning committed:
gsettings-desktop-schemas: 3.32.0 → 3.34.0

hedning committed:
gnome3.hitori: 3.32.0 → 3.34.0

hedning committed:
gnome3.iagno: 3.32.0 → 3.34.0

hedning committed:
libdazzle: 3.32.3 → 3.34.0

hedning committed:
gtksourceview4: 4.2.0 → 4.4.0

hedning committed:
libgweather: 3.32.2 → 3.33.92

hedning committed:
libpeas: 1.22.0 → 1.24.0

hedning committed:
gnome3.polari: 3.32.2 → 3.34.0

hedning committed:
gnome3.nautilus: 3.32.3 → 3.34.0

* add gobject-introspection for gir

Co-authored-by: worldofpeace <worldofpeace@protonmail.ch>

hedning committed:
gnome3.quadrapassel: 3.32.0 → 3.34.0

hedning committed:
orca: 3.32.0 → 3.34.0

hedning committed:
gnome3.sushi: 3.32.1 → 3.34.0

hedning committed:
simple-scan: 3.32.2.1 → 3.34.0

hedning committed:
gnome3.totem: 3.32.1 → 3.34.0

vala and nautilus support was removed in 3.32.

hedning committed:
gnome3.tracker: 2.2.2 → 2.3.0

hedning committed:
gnome3.tracker-miners: 2.2.2 → 2.3.0

hedning committed:
yelp: 3.32.2 → 3.34.0

hedning committed:
gnome3.yelp-xsl: 3.32.1 → 3.34.0

hedning committed:
jsonrpc-glib: 3.32.0 → 3.33.3

hedning committed:
atk: 2.32.0 → 2.34.1

Fix .pc file requires gobject:
https://gitlab.gnome.org/GNOME/atk/merge_requests/30

hedning committed:
vte: 0.56.3 → 0.58.0

Co-authored-by: worldofpeace <worldofpeace@protonmail.ch>

hedning committed:
networkmanager-iodine: fix build with glib-2.62

hedning committed:
gnome3.gnome-notes: 3.32.2 → 3.34.0

hedning committed:
gnome3.atomix: 3.32.1 → 3.34.0

hedning committed:
glib-networking: 2.60.3 → 2.62.0

hedning committed:
gnome3.gnome-boxes: 3.32.1 → 3.34.0

oVirt support was dropped

hedning committed:
gnome3.gnome-online-miners: 3.30.0 → 3.34.0

hedning committed:
gobject-introspection: 1.60.2 → 1.62.0

hedning committed:
gnome3.libgnome-games-support: 1.4.3 → 1.4.4

hedning committed:
gnome3.lightsoff: 3.32.0 → 3.34.0

hedning committed:
gnome3.swell-foop: 3.32.0 → 3.34.0

hedning committed:
gnome3.adwaita-icon-theme: 3.32.0 → 3.34.0

hedning committed:
gnome3.gnome-applets: 3.32.0 → 3.34.0

GSD no longer required since modem-lights applet was removed.

hedning committed:
gnome3.metacity: 3.32.0 → 3.34.0

hedning committed:
gdl: 3.28.0 → 3.34.0

hedning committed:
gspell: 1.8.1 → 1.8.2

hedning committed:
gnome3.anjuta: 3.28.0 → 3.34.0

hedning committed:
gnome3.devhelp: 3.32.0 → 3.34.0

hedning committed:
gnome3.gnome-software: 3.32.4 → 3.34.0

hedning committed:
baobab: 3.32.0 → 3.34.0

hedning committed:
gnome3.file-roller: 3.32.1 → 3.32.2

hedning committed:
gnome3.dconf: 0.32.0 -> 0.34.0

hedning committed:
gnome3.gnome-shell: 3.32.2 -> 3.34.0

hedning committed:
gnome3.mutter: 3.32.2 -> 3.34.0

hedning committed:
libsoup: 2.66.2 -> 2.68.1

hedning committed:
sysprof: 3.32.0 -> 3.34.0

hedning committed:
libmbim: 1.18.2 -> 1.20.0

fix build with glib 2.62

hedning committed:
libqmi: 1.22.4 -> 1.22.6

fix build with glib 2.62

hedning committed:
gnome3.aisleriot: 3.22.8 → 3.22.9

hedning committed:
gnomeExtensions.gsconnect: 23 -> 26

hedning committed:
dee: fix build with glib-2.62

worldofpeace commented:
cc @domenkozar @vcunat @NixOS/infra as we need the [jobset](https://hydra.nixos.org/jobset/nixpkgs/gnome#tabs-configuration) reconfigured with `gnome-3.34` as the target branch.

zimbatm commented:
@worldofpeace done

hedning committed:
vte-ng: fix build

vte now uses meson so don't bother overriding vte.

hedning committed:
gcolor3: fix build

hedning committed:
gnome3.gpaste: 3.32.0 -> 3.34.0

hedning committed:
gnucash: fix build with glib-2.62

hedning committed:
bamf: fix build with glib-2.62

hedning committed:
libunique: fix build with glib-2.62

hedning committed:
ldmtool: fix build with glib-2.62

hedning committed:
xlog: fix build with glib-2.62

hedning committed:
purple-matrix: fix build with glib-2.62

hedning committed:
openjfx12: fix build with glib-2.62

hedning committed:
openjfx11: fix build with glib-2.62

hedning committed:
putty: fix build with glib-2.62

hedning committed:
gnome3.gnome-backgrounds: 3.32.0 → 3.34.0

hedning committed:
grilo: 0.3.9 → 0.3.10

Dropped the meson_options patch, not sure if it was necessary.

hedning committed:
gnome3.grilo-plugins: 0.3.9 → 0.3.10

hedning commented:
atk caused some issues for a few dependencies as the pc file no longer Requires `gobject`. I added both `glib` and `gobject` as the patch was easier that way, and it shouldn't really be a problem as `gobject` itself require `glib`.

hedning commented:
In grilo I dropped the [meson_options patch](https://gitlab.gnome.org/GNOME/grilo/commit/f6993c2a8a6c1a6246372569f9f7a9179955c95e?merge_request_iid=45) as it no longer applied, not sure if did anything important? (the vapi stuff was included in the output at least)

jtojnar commented:
I will try to take a look during the working week.

BTW, I added a commit parsing to the changelog obtainer: https://github.com/jtojnar/what-changed#commits

hedning committed:
nixos/gdm: start after getty@tty1

GDM is now killed if tty1 is started after gdm is launched. This follows
upstream's gdm service config.

This might cause problems with nixos-rebuild switch though. See the reasoning
and work that led to not following upstream on this:
  https://github.com/NixOS/nixpkgs/issues/21439
  4a180da
  c46d4da

worldofpeace committed:
libsecret: 0.18.8 -> 0.19.1

worldofpeace committed:
gnome3.gnome-autoar: cleanup

* enable vapi
* multi outputs
* correct Inputs

worldofpeace committed:
gnome3.gnome-shell: enable extensions_tool

worldofpeace commented:
> It's easy to enable and works OK, but it looks like there's some environment trobles (eg. I'm getting the wrong fonts/theme etc.)

Is it possible you could try with the commits from https://github.com/NixOS/nixpkgs/pull/68492 @hedning?

Also about realtime scheduling https://github.com/NixOS/nixpkgs/issues/68318#issuecomment-531546645. I believe what you've suggested there is https://github.com/NixOS/nixpkgs/pull/68608/commits/20d87c44d9db4f9ed37d948a43d71ed8d4c9c2af?
It's the same sort of thing I did for gnome-keyring
* https://github.com/NixOS/nixpkgs/commit/2d6247a41477513d47cdec0e268c62296e1bb931

hedning commented:
> Is it possible you could try with the commits from #68492 @hedning?

Yeah, can give it a go

Seahorse doesn't [build with the new libsecret](https://gitlab.gnome.org/GNOME/seahorse/issues/242) btw.

> Relevant?

Yeah, testing this in a few configurations:
- starting gdm on tty1 and waiting for tty1 works, apart from nixos-rebuild switching (which might be unrelated https://github.com/NixOS/nixpkgs/issues/44344)
- starting gdm on tty7 and waiting for tty1 also works for some reason, but it broke gnome-shell after switching to a console and back (no idea if that's reproducible)
- starting gdm on tt7 and waiting on tty7 results in a tty jump when the gnome-session starts.

I'd say we should go for tty1 and having the conflict (as upstream does), though I guess we should look out for the rebuild issue: https://github.com/NixOS/nixpkgs/issues/21439 https://github.com/NixOS/nixpkgs/commit/4a180dac60e066f52268b69820bafa89dc4ba0d3 https://github.com/NixOS/nixpkgs/commit/c46d4dab967572be809f302aef36ed07d4976492

hedning committed:
nixos/gnome3: enable systemd session

- register gnome-session and gnome-settings-daemon services.
- gnome-shell is already registered due to having a xdg portal
- manually specify that gsd is wanted by gnome-session
    (systemd.packages doesn't pick the .wants directories for some reason)

hedning commented:
systemd session doesn't start gsd properly. This causes a bunch of problems like not picking up the correct theme etc. `gsd` in `systemd.packages` isn't sufficient for some reason though. Will investigate some more.

hedning committed:
gnome3.gnome-shell: hardcode gsettings path

hedning committed:
webkitgtk: 2.24.4 -> 2.26.1

* add fix-bubblewrap-paths.patch
* use clangStdenv
  We're running into #36947

Co-authored-by: worldofpeace <worldofpeace@protonmail.ch>

worldofpeace commented on nixos/modules/services/desktops/gnome3/gnome-settings-daemon.nix:
```diff
@@ -38,8 +38,22 @@ in
 
     environment.systemPackages = [ cfg.package ];
 
+    systemd.packages = [ cfg.package ];
```
So this module is shared by mate and pantheon, I don't think mate will have this config and pantheon still has to use an older g-s-d. Can we move this config to `services.gnome3.core-shell`, with a comment why it isn't in the g-s-d module?

hedning replied:
Yeah, that makes sense.

worldofpeace commented:
> Seahorse doesn't build with the new libsecret btw.

Don't know why there isn't a new release yet, I've updated to git master in the meantime.
There's also https://gitlab.gnome.org/GNOME/seahorse/issues/244 which I guess we would've bumped into.

hedning commented:
[Failed build log](https://gist.github.com/hedning/e3611ffbd9bf4a928c5c6d52afde9eca) of webkitgtk 2.26.0: https://github.com/NixOS/nixpkgs/commit/8b72d1979b1a81040eb48f9b646a6d00826ba414:

> lib/libWTFGTK.a(lib/../Source/WTF/wtf/CMakeFiles/WTF.dir/MediaTime.cpp.o):MediaTime.cpp:function WTF::MediaTime::setTimeScale(unsigned int, WTF::MediaTime::RoundingFlags): error: undefined reference to '__divmodti4'
collect2: error: ld returned 1 exit status

Not sure how to fix this (and the error occurs really late in the build annoyingly enough :(

worldofpeace committed:
gnome3.mutter: enable profiler

This needed a patch to get the correct directory to the sysprof
dbus interface.

worldofpeace commented:
I've enabled mutter's profiler https://github.com/NixOS/nixpkgs/pull/68608/commits/f1eb68393db0a7374379d914f76f68ad40527489, this needed a little patch. Haven't got to submitting it because GitLab seems to be having a hard time forking mutter for me :smile:

jtojnar commented on pkgs/desktops/gnome-3/core/mutter/0001-build-add-sysprof_dbus_inferfaces_dir-option.patch:
```diff
@@ -0,0 +1,52 @@
++  dbus_interfaces_dir = get_option('sysprof_dbus_interfaces_dir')
++
++  if dbus_interfaces_dir == ''
++    dbus_interfaces_dir = join_paths(datadir, 'dbus-1', 'interfaces')
```
Maybe something like this:

```patch
--- a/src/meson.build
+++ b/src/meson.build
@@ -710,7 +710,7 @@
     'backends/meta-profiler.h',
   ]
 
-  dbus_interfaces_dir = join_paths(datadir, 'dbus-1', 'interfaces')
+  dbus_interfaces_dir = join_paths(sysprof_dep.get_pkgconfig_variable('datadir'), 'dbus-1', 'interfaces')
   sysprof3_dbus_file = join_paths(dbus_interfaces_dir, 'org.gnome.Sysprof3.Profiler.xml')
 
   dbus_sysprof3_profiler_built_sources = gnome.gdbus_codegen('meta-dbus-sysprof3-profiler',
```

worldofpeace replied:
`sysprof-capture-3.pc` doesn't have `datadir`.

worldofpeace replied:
`.pc` files look a bit funky too
```
Libs: -L${prefix}//nix/store/3snygqmijn2in3k6n402sv1ha76mw1gy-sysprof-3.34.0-lib/lib  -lsysprof-capture-3
```

jtojnar replied:
Maybe prefix then?

Bah, another hand-crafted pc file?

hedning replied:
strange, looks like it's made with `pkgconfig.generate`: https://gitlab.gnome.org/GNOME/sysprof/blob/sysprof-3.34.0/src/libsysprof-capture/meson.build#L58-67

worldofpeace replied:
Did https://github.com/NixOS/nixpkgs/pull/68608/commits/931aa0e5e2b9a4699a00abdee723dac0e0af8242, far to lazy to make sysprof idiomatic :smile:

jtojnar replied:
`get_option('datadir')` will probably return absolute path, since we are using multiple outputs.

worldofpeace replied:
It seemed to work, doing `'share'` feels kinda dirty at this point.

jtojnar replied:
Oh, we are currently not setting `datadir`. But I still would not rely on the default value here.

jtojnar commented:
For Seahorse, we should probably switch to https://github.com/mesonbuild/meson/pull/5213

hedning commented:
`webkitgtk-2.26` builds with clangStdenv, not sure if that's a reasonable option though?

worldofpeace commented:
> `webkitgtk-2.26` builds with clangStdenv, not sure if that's a reasonable option though?

I'd say it isn't the only expression that has to do this. Though I'd also say my opinion on this issue particular isn't extremely valuable :rofl:

hedning committed:
networkmanager_strongswan: fix build with glib-2.62

hedning committed:
stupidterm: fix build with updated vte

vte no longer propagates pcre2

hedning committed:
xfce4-14.xfce4-terminal: fix build with updated vte

vte no longer propagates pcre2.

hedning committed:
deepin.deepin-terminal: fix build with updated vte

vte no longer propagates pcre2 and gnutls

hedning committed:
roxterm: fix build with updated vte

vte no longer propagates pcre2

hedning committed:
planner: fix build with glib-2.62

hedning committed:
pidgin-sipe: fix build with glib-2.62

hedning committed:
sakura: fix build with updated vte

vte no longer propagates pcre2.

worldofpeace committed:
gnome3.gnome-todo: fix build with libecal 2.0

worldofpeace committed:
gnome3.nautilus-sendto: remove

Upstream doesn't recommended its use/ it's archived [0]

[0]: https://github.com/GNOME/gnome-build-meta/commit/ba9f5e257cd430d01e7c53aaed8e9a26975760e6

worldofpeace committed:
pantheon.elementary-calendar: 5.0 -> 2019-09-17

This is needed for the libical 2.0 compat that landed in master [0]

[0]: https://github.com/elementary/calendar/issues/404

worldofpeace commented:
Opened MRs upstream for the undclared gio-unix-2.0 
* https://gitlab.gnome.org/GNOME/gnome-builder/merge_requests/215
* https://gitlab.gnome.org/GNOME/gnome-user-share/merge_requests/15

Not yet for gnome-terminal. Rather interesting for that one, there wasn't that problem before IIRC.

hedning committed:
lxqt-archiver: 0.0.96 -> 2019-09-15

fix build with glib-2.62

worldofpeace committed:
gnome3.gnome-sound-recorder: 3.32.1 -> 3.34.0

dtzWill committed:
gtk3: 3.24.10 -> 3.24.11

worldofpeace committed:
glibmm: 2.60.0 -> 2.62.0

worldofpeace commented:
Perhaps we should rebase given we're rebuilding for gtk3?

worldofpeace committed:
gdk-pixbuf: 2.38.1 → 2.38.2

worldofpeace commented:
Rebased on https://hydra.nixos.org/eval/1542887#tabs-inputs eval.
I also noticed that we were using a gdk-pixbuf snapshot release and not a stable release.
Fixed in https://github.com/NixOS/nixpkgs/pull/68608/commits/b75b26d5b38eabbcd3ccb695dda48b43928f4091

worldofpeace commented:
We might want to comment on upstream atk since we have a patch to move glib to `Requires`
* https://gitlab.gnome.org/GNOME/atk/merge_requests/30

worldofpeace committed:
vala_0_44: 0.44.6 -> 0.44.7

worldofpeace committed:
vala_0_38: remove

worldofpeace committed:
vala_0_36: 0.36.19 -> 0.36.20

worldofpeace committed:
vala_0_46: init at 0.46.1

worldofpeace committed:
vala_0_42: remove

worldofpeace committed:
vala: use vala_0_46

worldofpeace committed:
dee: pull in patches

Fixes the deprecation warnings and vapi stuff.
This also allows us to use latest vala.

worldofpeace committed:
umockdev: use latest vala

worldofpeace commented:
@NixOS/infra @domenkozar Could any of you give the jobset 100 more scheduling shares?
Currently it's at 200 and I'd think 300 would be a better spot like it was before.

worldofpeace committed:
gnome3.gnome-initial-setup: init at 3.34.0

worldofpeace committed:
nixos/gnome-initial-setup: init

It tries to start during the GDM session even
with the conflicts.

worldofpeace commented:
Added gnome-initial-setup module https://github.com/NixOS/nixpkgs/pull/68608/commits/78a849b82d235e51889d847fb6af507dcbda1967.
There's an issue with it trying to start in GDM. Upstream has conflicts for this which I've manually setup in the module, but they seem to have no affect.

jtojnar commented on pkgs/desktops/gnome-3/core/gnome-initial-setup/default.nix:
```diff
@@ -0,0 +1,115 @@
+    "-Dsoftware-sources=disabled"
+    "-Dibus=disabled"
+    # Disable pages not right for NixOS
+    # For example user accounts should be preconfigured
```
Perhaps the comment should go to the file itself.

worldofpeace replied:
I wasn't sure if it was smart enough to ignore comments. Will check.

jtojnar replied:
I would expect them to use keyfile parser fromg GLib, which should support comments.

worldofpeace committed:
gnome3.gdm: add branding

We use the logo key to add the NixOS text logo to GDM.

worldofpeace committed:
nixos/gdm: prevent g-i-s from running

worldofpeace committed:
nixos/gnome-initial-setup: prevent run on existing systems

GNOME initial setup's run is conditioned on whether
the gnome-initial-setup-done file exists in XDG_CONFIG_HOME
Because of this, every existing user will have initial setup
running because they never ran it before.

To prevent this we create the file if the users stateVersion
is older than 20.03 (the release we added this module).

Also drop uneeded manual conflicts as systemd.packages
does handle this.

worldofpeace commented on nixos/modules/services/desktops/gnome3/gnome-initial-setup.nix:
```diff
@@ -4,6 +4,44 @@
+  # running because they never ran it before.
+  #
+  # To prevent this we create the file if the users stateVersion
+  # is older than 20.03 (the release we added this module).
```
Indeed this is terrifying :smile: 

No idea why this wasn't rewritten to use gsettings, Pantheon's onboarding application keeps a key of pages viewed to condition it starting. With something like that this could become a simple gsettings override.

worldofpeace committed:
epiphany: 3.32.4 -> 3.34.0

worldofpeace committed:
libunity: fix build with latest vala

See: https://gitlab.gnome.org/GNOME/vala/issues/766

worldofpeace committed:
feedreader: fix build with latest libsecret

worldofpeace committed:
midori: 7 -> 9.0

* add libarchive and json-glib

We use vala_0_44 to avoid the issue with sqlite3 vapi [0]

[0]: https://gitlab.gnome.org/GNOME/vala/commit/a54e4d89a449044c965869abf0ce78dadcf093ac.patch

worldofpeace committed:
libpeas: correct gtk .pc file

Here's the diff post meson:
-prefix=/nix/store/hfrrby8wapcz17v927as18z3znqdv2ps-libpeas-1.22.0
-exec_prefix=${prefix}
-libdir=${exec_prefix}/lib
+prefix=/nix/store/sc8h3m6fv3zdjzpkimhm4yqarafhjp6y-libpeas-1.24.0
+libdir=${prefix}/lib
 includedir=${prefix}/include

-Name: libpeas-gtk
-Description: libpeas-gtk, a GObject plugins library (Gtk widgets)
-Requires: glib-2.0 >= 2.38.0, gtk+-3.0 >= 3.0.0, libpeas-1.0 >= 1.22.0
-Version: 1.22.0
+Name: libpeas-gtk-1.0
+Description: GObject plugins library widgetery
+Version: 1.24.0
+Requires.private: libpeas-1.0, glib-2.0 >=  2.38.0, gobject-2.0 >=  2.38.0, gobject-introspection-1.0 >=  1.39.0, gtk+-3.0 >=  3.0.0, gmodule-2.0 >=  2.38.0, gio-2.0 >=  2.38.0
+Libs: -L${prefix}//lib  -lpeas-gtk-1.0
 Cflags: -I${includedir}/libpeas-1.0
-Libs: -L${libdir} -lpeas-gtk-1.0

As we can see, all the requires got moved into private.
This appears to have broken midori [0]

[0]: https://hydra.nixos.org/build/101627205

worldofpeace commented:
It appears that the meson port on libpeas caused some issues with midori.
I made it similar to how it was before with https://github.com/NixOS/nixpkgs/pull/68608/commits/53bfd2c7ec149a2f7e2b150db5657760419cbe38.

hedning commented:
Gnome-shell gets `cap_sys_nice+eip` when run, not just `+ep`. This looks like a limitation with the nixos security wrappers (ambient -> inheritable): https://github.com/NixOS/nixpkgs/blob/53bfd2c7ec149a2f7e2b150db5657760419cbe38/nixos/modules/security/wrappers/wrapper.c#L226-L229

In particular this makes launching epiphany from gnome-shell crash here (bwrap do not want to act when it's not setuid, but have caps set)
https://github.com/containers/bubblewrap/blob/93047df6052034989a576de55a6fe2cce05f3524/bubblewrap.c#L762-L768

Other than that I think 3.34 is ready to go :)

worldofpeace committed:
gnomeExtensions.mediaplayer: remove

Source repo is archived and the extension was retired [0]

[0]: https://github.com/JasonLG1979/gnome-shell-extensions-mediaplayer/blob/master/README.md

worldofpeace committed:
gnomeExtensions.appindicator: 29 -> 30

worldofpeace committed:
gnomeExtensions.arc-menu: 31 -> 33.2

worldofpeace committed:
gnomeExtensions.caffeine: make as broken

worldofpeace committed:
gnomeExtensions.clipboard-indicator: mark as broken

worldofpeace committed:
gnomeExtensions.icon-hider: mark as broken

worldofpeace committed:
gnomeExtensions.pidgin-im-integration: mark as broken

worldofpeace committed:
gnomeExtensions.remove-dropdown-arrows: 11 -> 13

worldofpeace committed:
gnomeExtension.sound-output-device-chooser: 2019-04-01 -> 24

worldofpeace committed:
gnomeExtension.system-monitor-applet: mark as broken

worldofpeace committed:
gnomeExtension.taskwhisperer: mark as broken

worldofpeace committed:
gnomeExtension.tilingnome: 2019-01-18 -> 2019-09-19

worldofpeace committed:
gnomeExtension.timepp: mark as broken

worldofpeace committed:
gnome3.pomodoro: 0.15.1 -> 2019-09-15

hedning commented:
Launching flatpaks from gnome-shell doesn't work with inherited capabilities either. We probably need to drop 3fa3005298a55f61fa71b56af6d7134ec178ed33 and tackle it in another PR tbh.

worldofpeace commented:
> Launching flatpaks from gnome-shell doesn't work with inherited capabilities either. We probably need to drop [3fa3005](https://github.com/NixOS/nixpkgs/commit/3fa3005298a55f61fa71b56af6d7134ec178ed33) and tackle it in another PR tbh.

Fair, I'll make it a separate feature after we merge.

hedning commented:
`gnome-music` and `totem` crashes on startup, reverting `grilo-plugins` to 0.3.9 fixes the issue. Both programs crash in `insert_key_mapping () from /nix/store/ldzbj98xp3z0phbmmp7drvf49xrdrsf1-grilo-plugins-0.3.10/lib/grilo-0.3/libgrltracker.so`.

worldofpeace committed:
pantheon.elementary-settings-daemon: fix gnome-session detecting autostarts

For some reason GNOME 3.34 gnome-session fails to recognize
these autostarts suffixed with `-pantheon`, whereas they worked before.
It seems to work fine without the suffix, and it's verified elementary-settings-daemon
proccess are running in the session.

worldofpeace committed:
gnome3.gdm: fix rebuild switch

See: https://gitlab.gnome.org/GNOME/gdm/issues/515

hedning committed:
gnome3.gnome-shell: drop caribou

This should no longer be necessary as gnome-shell ships its own on-screen
keyboard.

worldofpeace committed:
gnome3.gdm: Remove duplicate sessions when on Xorg

worldofpeace committed:
gnome3.gnome-session: fix debug spam with systemd session

worldofpeace committed:
nixos/gnome3: use sessionVariables

Without this GNOME_SESSION_DEBUG
likely won't be set in the systemd user environment.

worldofpeace commented on pkgs/desktops/gnome-3/core/gnome-session/default.nix:
```diff
@@ -12,6 +12,17 @@ stdenv.mkDerivation rec {
+      sha256 = "0vjg77gpj3k63a5ffhsvv5m30lbj6cab35lhl4gpqxg4j2j3yy7y";
+    })
+    (fetchpatch {
+      url = "https://gitlab.gnome.org/GNOME/gnome-session/commit/b7b24627485c520f873db4e918e217a76ededd8c.patch";
```
This patch borrows some code already in gdm that's for unsetting variables.
Is it possible we need to add 
* https://github.com/NixOS/nixpkgs/blob/master/pkgs/desktops/gnome-3/core/gdm/reset-environment.patch
as well?

hedning replied:
Not entirely sure. This happens after [`bin/gnome-session`](https://gitlab.gnome.org/GNOME/gnome-session/blob/master/gnome-session/gnome-session.in), which sets up the environment.  So at that point we want `__NIXOS_SET_ENVIRONMENT_DONE` to be propagated (eg. when running shells inside the session).

hedning replied:
Yeah, I'm unable to reproduce the problem by logging in and out from gnome-shell while keeping `systemd --user` alive.

worldofpeace commented:
With https://github.com/NixOS/nixpkgs/pull/68608/commits/02f62a34ea09d0b12a704c47d9d530dec16771e6 as a downstream patch rebuild switch appears to be fixed :sparkle: I've checked and others have adopted it pretty quick.

worldofpeace commented:
I have pretty much no clue why gnome-music/totem crash, the only change they rounded up was
* https://gitlab.gnome.org/GNOME/tracker/blob/2.3.0/NEWS#L6

and they migrated to that properly everywhere.

hedning committed:
gnome3.mutter: Add upstream patches before 3.34.1

Fixes some eg. animation related crashes which in particular affected
dash-to-dock.

worldofpeace commented on pkgs/desktops/gnome-3/core/yelp-xsl/default.nix:
```diff
@@ -10,14 +17,22 @@ stdenv.mkDerivation rec {
+  passthru = {
+    updateScript = gnome3.updateScript {
+      packageName = "yelp-xsl";
+      attrPath = "gnome3.yelp-xsl";
```
pname?

jtojnar committed:
vte-ng: move inside termite

It is just a few patches only needed by termite. Let's not pollute our tree with it.

jtojnar committed:
gnome3.seahorse: fix build with recent libsecret

jtojnar committed:
grilo-plugins: build grl-bookmarks

Totem asks after it.

worldofpeace commented:
I wonder if unsetting `NO_AT_BRIDGE=1` could cause problems on hydra @jtojnar . 
I recall there being issues there when it wasn't set

jtojnar commented:
> I wonder if unsetting `NO_AT_BRIDGE=1` could cause problems on hydra @jtojnar .
> I recall there being issues there when it wasn't set

I based this on the following lines in changelogs:

* https://gitlab.gnome.org/GNOME/libdazzle/blob/3.34.0/NEWS#L7
* https://gitlab.gnome.org/GNOME/gtksourceview/blob/4.4.0/NEWS#L3

worldofpeace committed:
gnome3.evolution: fix FHS paths

This includes commands to spamassassin, sa-learn, bogofilter
and the prefix to openldap

worldofpeace commented:
Yep the set the required environment :+1: 
* https://gitlab.gnome.org/GNOME/libdazzle/blob/3.34.0/tests/meson.build#L10
* https://gitlab.gnome.org/GNOME/gtksourceview/blob/4.4.0/testsuite/meson.build#L7

worldofpeace commented:
I checked the spam filtering stuff with
* https://help.ubuntu.com/community/HowToEnableSpamFilteringInEvolution

As it's still up to date Ubuntu documentation (scratches head), we'll just say evolution doesn't change much :smile: 
> (last edited 2008-10-12 23:51:13)

I strangely couldn't find the plugin in the list, but I still could select either in the Junk settings.
Also checked the right path was even in the ELF.

worldofpeace committed:
pantheon.wingpanel-indicator-datetime: use older e-s-d

Workaround https://github.com/elementary/wingpanel-indicator-datetime/issues/122
Revert once resolved upstream.

jtojnar committed:
grilo-plugins: add grl-chromaprint

worldofpeace commented on pkgs/development/libraries/grilo-plugins/default.nix:
```diff
@@ -36,6 +38,21 @@ stdenv.mkDerivation rec {
+    # * playbin (gst-plugins-base)
+    # * chromaprint (gst-plugins-bad)
+    (substituteAll {
+      src = ./chromaprint-gst-plugins.patch;
```
need to stage the patch, (though it's wip anyway)

hedning committed:
gnomeExtensions.dash-to-dock: 66 -> 20190921

There's still no official release for gnome-3.34

hedning commented on pkgs/desktops/gnome-3/core/mutter/0001-build-use-get_pkgconfig_variable-for-sysprof-dbusdir.patch:
```diff
@@ -0,0 +1,25 @@
+   ]
+ 
+-  dbus_interfaces_dir = join_paths(datadir, 'dbus-1', 'interfaces')
++  dbus_interfaces_dir = join_paths(sysprof_dep.get_pkgconfig_variable('prefix'), 'share', 'dbus-1', 'interfaces')
```
Switched to 'share' in case we set `datadir` in the future.

hedning commented:
I'll push to staging soon if no one has any complaints :)

worldofpeace commented:
@hedning I believe @jtojnar raised some concerns in #nixos-dev yesterday.

The main one was
> * glib: Add various installed utilities’ paths to `gio-2.0.pc` (#1796) https://gist.github.com/jtojnar/9c008fa1da64f2fbfab9a42aa0f6e3c5

`gio-2.0.pc` has the [utilities](https://gist.github.com/worldofpeace/6d742389548afdd697ddb40e15482a67) in there and the paths are incorrect.
Really wishing it doesn't have to be a downstream patch for our outputs setups once again.

worldofpeace committed:
libproxy: build with spidermonkey_60

hedning commented:
Is there any closure reduction in having a special `glib.bin` which isn't `bindir`? Else I'm not sure if it's worth to fight upstream on this? Having a few binaries less in the user path isn't a huge win in my opinion at least.

(an automatic pkgconfig checker would be pretty handy).

jtojnar commented:
Most of the binaries in `dev` are Python so we do not want to have them in `out`.

jtojnar commented:
Checking the output with diffoscope looks fine.

jtojnar commented:
c4d8173dedcc9d5ce8a92d658db4533befde073a should be squashed into c64069d8ffb4a968c93c65c60d805d61f16591a8.

Also master should be fine, since we are already building everything.

hedning commented:
Right, could've worded that better. Is there any problem putting all the binaries in `glib.bin`, that should serve the same purpose as putting some of them in `dev` right? (apart from exposing some perhaps not super useful binaries to the users path)

hedning commented:
> Also master should be fine, since we are already building everything.

Haven't rebased on master in a while, though we could do that now since we're rebuilding glib.

jtojnar commented:
> Right, could've worded that better. Is there any problem putting all the binaries in `glib.bin`, that should serve the same purpose as putting some of them in `dev` right? (apart from exposing some perhaps not super useful binaries to the users path)

Some packages depend on things in `glib.bin` but not on Python. This would increase closure for them. But maybe it is reasonable cost for simplification of `glib` package.

hedning commented:
Before I rebase, there's quite a few merge conflicts in staging due to the `hicolor-icon-theme` removal, so that's not ideal for the people who'll have to resolve them.

jtojnar commented:
I will do a m→s merge after this gets to master.

hedning commented:
squashed the glib patch, lets handle a potential move to having all binaries in `glib.bin` later.

worldofpeace committed:
gnome3.evince: 3.32.0 -> 3.34.0

worldofpeace committed:
gnome3.gnome-keyring: 3.31.91 -> 3.34.0

worldofpeace commented:
@GrahamcOfBorg eval

jtojnar commented:
Past merge times:
* 3.24: https://github.com/NixOS/nixpkgs/pull/26879 & https://github.com/NixOS/nixpkgs/pull/28053, 64 days
* 3.26: https://github.com/NixOS/nixpkgs/pull/29392, 76 days (Meson rpath scandal)
* 3.28: https://github.com/NixOS/nixpkgs/pull/36893, 9 days
* 3.30: https://github.com/NixOS/nixpkgs/pull/45950, 89 days (Meson broke everything again)
* 3.32: https://github.com/NixOS/nixpkgs/pull/57027, 29 days
* 3.34: https://github.com/NixOS/nixpkgs/pull/68608, 20 days

worldofpeace commented:
So I've added a GitHub project for the next release ahead of time
* https://github.com/NixOS/nixpkgs/projects/24

Organizationally, I don't find it convenient for us to edit the issue when there's GitHub projects for this.

FelschR commented:
So, as a NixOS newbie can someone tell me if / how I can update to GNOME 3.34 from nixos-unstable when I'm running 19.09?

hedning commented:
@FelschR your best bet is running nixos-unstable to be honest, it's quite stable tbh, and if something actually breaks there's rollback :)

With `nix-channel` you should be able to switch like this:
```bash
nix-channel --add https://nixos.org/channels/nixos-unstable nixos
nixos-rebuild boot --upgrade
```

If you want some more manual control, you can build from a checked out git repo (the channels remote , `git://github.com/NixOS/nixpkgs-channels.git`, reflects the revision you get with `nix-channel`):
```bash
cd ~/path/to/nixpkgs
git checkout channels/nixos-unstable # or whatever commit you want :)
nixos-rebuild boot -I nixpkgs=./.
```

Hope that helps :)

jtojnar commented:
@FelschR unfortunately, desktop apps often depend on many systemd/D-Bus services, and since those environments only supports a single namespace, it would conflict with your running system (unless you used some kind of ultra-minimal configuration). For that reason there is no supported way to run GNOME 3.34 on 19.09.

I suggest you switch your whole system to `nixos-unstable`, despite the name it is quite stable, since we require tests to pass before bumping the channel.

FelschR commented:
@jtojnar Thank you for the quick answer and insightful explanation.
I'll consider making the channel switch.

nixos-discourse commented:
This pull request has been mentioned on **NixOS Discourse**. There might be relevant details there:

https://discourse.nixos.org/t/nixos-20-03-feature-freeze/5655/1
